azure/ConsiderationsForServiceDesign.md

```diff
@@ -0,0 +1,97 @@
+
+Azure Service teams should engage the  Azure HTTP/REST Stewardship Board early in the development lifecycle for guidance, discussion, and review of their API. In addition, it is good practice to perform a security review, especially if you are concerned about PII leakage, compliance with GDPR, or any other considerations relative to your situation.
+
+Your goal is to create a developer friendly API where:
+
+> :white_check_mark: **DO** ensure that customer workloads never break
+> 
+> :white_check_mark: **DO** ensure that customers are able to adopt a new version of service or SDK client library <b>without requiring code changes</b> 
+
+### Azure Management Plane vs Data Plane
+*Note: Developing a new service requires the development of at least 1 (management plane) API and potentially one or more additional (data plane) APIs.  When reviewing v1 service APIs, we see common advice provided during the review.*
+
+A **management plane** API is implemented through the Azure Resource Manager (ARM) and is used by subscription administrators. A **data plane** API is used by developers to implement applications. Occasionally, some operations are useful to both administrators and developers. In this case, the operation can appear in both APIs. Although, best practices and patterns described in this document apply to all HTTP/REST APIs, they are especially important for **data plane** services because it is the primary interface for developers using your service. The **management plane** APIs may have other preferred practices based on the conventions of the Azure ARM.
+
+### Start with the Developer Experience
+A great API starts with a well thought out and designed service. Your service should define simple/understandable abstractions with each given a clear name that you use consistently throughout your API and documentation. There must also be an unambiguous relationship between these abstractions.
+
+It is extremely difficult to create an elegant API that works well on top of a poorly designed service; the service team and customers will live with this pain for years to come. So, the service team should empathize with customers by:
+ - Building apps that consume the API
+ - Hold reviews and share what is learned with your team
+ - Get customer feedback from API previews
+ - Thinking about the code that a customer writes both before and after an HTTP operation
+ - Initializing and reading from the data structures your service requires
+ - Thinking about which errors are recoverable at runtime as opposed to indicating a bug in the customer code that must be fixed
+
+The whole purpose of a preview to address feedback by improving abstractions, naming, relationships, API operations, and so on. It is OK to make breaking changes during a preview to improve the experience now so that it is sustainable long term. 
+
+> :white_check_mark: **DO** provide examples in multiple languages
```

Comment:
I need some clarity here, since so far we didn't say anything about SDK creation and it seems we were really high level "this is what you should consider as guiding principles when coding the server". Which seems far from SDK at this point of the document.
Do we ask for WoZ fake pseudo-code? Are we asking them mostly to feel how the SDK could look like?

I'm fine with that approach to envision SDK impact right at the creation of the RestAPI, I just wonder if we should clarify the expectations of how to write the examples at this point.

Reply:
Updated the guidance and moved them to the hero scenario section. May not be the best fit, but feels a bit better.